junifer/markers/reho/reho_estimator.py

```diff
@@ -0,0 +1,551 @@
+        # Get scan data
+        niimg_data = data.get_fdata()
+        # Get scan dimensions
+        n_x, n_y, n_z, _ = niimg_data.shape
+
+        # Get rank of every voxel across time series
+        ranks_niimg_data = rankdata(niimg_data, axis=-1)
+
+        # Initialize 3D array to store tied rank correction for every voxel
+        tied_rank_corrections = np.zeros((n_x, n_y, n_z), dtype=np.float64)
+        # Calculate tied rank correction for every voxel
+        for i_x, i_y, i_z in product(range(n_x), range(n_y), range(n_z)):
+            # Calculate tied rank count for every voxel across time series
+            _, tie_count = np.unique(
+                ranks_niimg_data[i_x, i_y, i_z, :],
+                return_counts=True,
+            )
+            # Calculate and store tied rank correction for every voxel across
+            # timeseries
+            tied_rank_corrections[i_x, i_y, i_z] = np.sum(
+                tie_count**3 - tie_count
+            )
+
+        # Initialize 3D array to store reho map
+        reho_map = np.ones((n_x, n_y, n_z), dtype=np.float32)
+
+        # Calculate whole brain mask
+        mni152_whole_brain_mask = nmask.compute_brain_mask(
+            data, threshold=0.5, mask_type="whole-brain"
+        )
+        # Convert 0 / 1 array to bool
+        logical_mni152_whole_brain_mask = (
+            mni152_whole_brain_mask.get_fdata().astype(bool)
+        )
+
+        # Create mask cluster
+        if nneigh in (7, 19, 27):
+            mask_cluster = np.ones((3, 3, 3))
+
+            if nneigh == 7:
+                mask_cluster[0, 0, 0] = 0
+                mask_cluster[0, 1, 0] = 0
+                mask_cluster[0, 2, 0] = 0
+                mask_cluster[0, 0, 1] = 0
+                mask_cluster[0, 2, 1] = 0
+                mask_cluster[0, 0, 2] = 0
+                mask_cluster[0, 1, 2] = 0
+                mask_cluster[0, 2, 2] = 0
+                mask_cluster[1, 0, 0] = 0
+                mask_cluster[1, 2, 0] = 0
+                mask_cluster[1, 0, 2] = 0
+                mask_cluster[1, 2, 2] = 0
+                mask_cluster[2, 0, 0] = 0
+                mask_cluster[2, 1, 0] = 0
+                mask_cluster[2, 2, 0] = 0
+                mask_cluster[2, 0, 1] = 0
+                mask_cluster[2, 2, 1] = 0
+                mask_cluster[2, 0, 2] = 0
+                mask_cluster[2, 1, 2] = 0
+                mask_cluster[2, 2, 2] = 0
+
+            elif nneigh == 19:
+                mask_cluster[0, 0, 0] = 0
+                mask_cluster[0, 2, 0] = 0
+                mask_cluster[2, 0, 0] = 0
+                mask_cluster[2, 2, 0] = 0
+                mask_cluster[0, 0, 2] = 0
+                mask_cluster[0, 2, 2] = 0
+                mask_cluster[2, 0, 2] = 0
+                mask_cluster[2, 2, 2] = 0
+
+            # Convert 0 / 1 array to bool
+            logical_mask_cluster = mask_cluster.astype(bool)
+
+            for i, j, k in product(
```

Comment:
This can be written only once (no need for 125 neigh separately)

Just set:
```
st = 1
end = 2
if nneigh == 125:
   st = 2
   end = 3
```
end then index as `i - st : i + end`, with the product as `range(st, n_x - (end - 1)`